templates/cmd/main.go.tpl

```diff
@@ -127,4 +132,5 @@ func main() { //nolint: funlen // Why: We can't dwindle this down anymore withou
 	if err := async.RunGroup(acts).Run(ctx); err != nil {
 		log.Warn(ctx, "shutting down service", events.NewErrorInfo(err))
 	}
+  exitCode = 0
```

Comment:
This will always be hit? Maybe `return` in L133?

Reply:
it won't "always be hit" because initialization errors should shallow return but i do agree we should return and most likely error on 133 as opposed to just warn

we should, if `Run` doesn't already, filter out the context cancellation error though

Reply:
* We should check if the error returned by RunGroup(acts).Run(ctx) is context.Canceled, if so, don't trigger the `log.Warn`
* Change the `log.Warn` to an error, this is an error and then immediately return (so exitCode is still 1)
* Add a log below the current conditional saying the service is shutting down (normally, since that'd be a context.Canceled or natural shutdown

Reply:
yeah that encapsulates the conversation + i would just say as a nit for the log below the conditional maybe say sometihng like
> graceful shutdown process successful